routing/missioncontrol.go

```diff
@@ -393,26 +393,30 @@ func (m *MissionControl) GetHistorySnapshot() *MissionControlSnapshot {
 // future probability estimates. It returns a bool indicating whether this error
 // is a final error and no further payment attempts need to be made.
 func (m *MissionControl) ReportPaymentFail(rt *route.Route,
-	failureSourceIdx int, failure lnwire.FailureMessage) (bool,
+	failureSourceIdx *int, failure lnwire.FailureMessage) (bool,
 	channeldb.FailureReason) {
 
-	var (
-		failureVertex route.Vertex
-	)
+	var failureSourceIdxInt int
+	if failureSourceIdx == nil {
```

Comment:
this needs explanation/comments

Reply:
comments got lost when I moved this code. readded